src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const result = [];
+
+  actions.forEach((elem) => {
+    const { type, extraData, keysToRemove } = elem;
+
+    switch (type) {
+      case 'addProperties':
+        copyState = Object.assign(copyState, extraData);
+
+        result.push({ ...copyState });
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete copyState[key];
```

Comment:
The `delete` operator is used correctly to remove properties from `copyState`. Ensure that `keysToRemove` is always an array to avoid runtime errors.